### ISSUE TITLE
NativeAOT support

### ISSUE THREAD

**Issue by Armynator** [ProwlEngine/Prowl#181]
**Is your feature request related to a problem? Please describe.**
The current runtime build size is quite big and requires the .NET runtime. NativeAOT builds don't require the runtime and trim down the final binary size quite a lot.

**Describe the solution you'd like**
Add support for NativeAOT release builds, at least for built games. The editor requiring the .NET SDK is fine.

**Additional context**
I've tested building it with NativeAOT myself, as Silk.NET already supports it. While it works after registering the GLFW window and input platforms manually, the render pipeline can't initialize for some reason. Probably because something important related to it was trimmed out. Would be nice to get easy NativeAOT support out of the box.

**Comment by michaelsakharov** [comment]
There is a bit of research that needs to go into this
For example how the Serializer with all its reflection will handle itself.
And how do we handle cases of serializing generic types, like List<T> in NativeAOT

This is definitely something we should look into eventually though, As I believe it is required for Web, and possibly other platforms as well

**Comment by sinnwrig** [comment]
Build pipeline should now be AOT-compatible and `Assembly.LoadFrom` is gone from the player. However, there is likely to be some trimming issues and `DllImport`'s still in our way, specifically in `Prowl.Veldrid`. Will need to look into this more.

**Comment by ivan-bys** [comment]
any news on this? should we give this a spin with current LLM technology?

**Comment by michaelsakharov** [comment]
> any news on this? should we give this a spin with current LLM technology?

I think Prowl.Echo, Our Serialization library may be a Hurdle for NativeAOT due to Heavy Reflection usage? But I haven't investigated NativeAOT since I made this issue two years ago, So a lot may have changed.

It may be worth investigating again, and it never hurts to throw AI at it to see what it does 🤷

**Comment by sinnwrig** [comment]
@michaelsakharov as long as Echo doesn't do any IL emission it should work if we disable trimming on publish. I think pretty much everything should be mostly there, but there might be a bit of tweaking we need to do for reflection that assumes it can do type instantiation like `Activator.CreateInstance`

**Comment by ZedDevStuff** [comment]
Could also use source generators. All serialization libraries use it either for NativeAOT support or performance. Maybe both

**Comment by icavalheiro** [comment]
Yeah, I think a good mix of source generators should be good, again, we would would support AOT only for deployables, the editor itself shouldnt use it.
